app/assets/javascripts/govuk_publishing_components/analytics-ga4/gtm-link-click-tracker.js

```diff
@@ -0,0 +1,156 @@
+        for (var key in clickData) {
+          if (schema.event_data[key]) {
+            schema.event_data[key] = clickData[key]
+          }
+        }
+
+        this.trackClickEvent(schema)
+      }
+    },
+
+    getClickType: function (event) {
+      switch (event.type) {
+        case 'click':
+          if (event.ctrlKey) {
+            return 'ctrl click'
+          } else if (event.metaKey) {
+            return 'command/win click'
+          } else {
+            return 'primary click'
+          }
+        case 'mousedown':
+          return 'middle click'
+        case 'contextmenu':
+          return 'secondary click'
+      }
+    },
+
+    handleMousedown: function (event) {
+      // 1 = middle mouse button
+      if (event.button === 1) {
+        this.handleClick(event)
+      }
+    },
+
+    isMailToLink: function (href) {
+      if (!href) {
+        return false
+      }
+
+      if (href.length < 7) {
+        return false
+      }
+      return href.substring(0, 7) === 'mailto:'
+    },
+
+    isDownloadLink: function (href) {
+      var assetsDomain = 'assets.publishing.service.gov.uk/'
+      var uploadsPath = '/government/uploads/'
+      if (!href) {
+        return false
+      }
+
+      if (this.hrefPointsToDomain(href, assetsDomain)) {
+        return true
+      }
+
+      var isInternalLink = this.hrefPointsToDomain(href, this.internalLinksDomain) || this.hrefPointsToDomain(href, this.internalLinksDomain.replace('www.', ''))
+      if (isInternalLink && href.indexOf(uploadsPath) !== -1) {
+        return true
+      }
+
+      // Checks relative links to the uploadsPath
+      if (this.stringStartsWith(href, uploadsPath)) {
+        return true
+      }
+    },
+
+    isExternalLink: function (href) {
+      if (!href) {
+        return false
+      }
+      var isInternalLink = this.hrefPointsToDomain(href, this.internalLinksDomain) || this.hrefPointsToDomain(href, this.internalLinksDomain.replace('www.', ''))
+      if (!isInternalLink && !this.hrefIsRelative(href)) {
+        return true
+      }
+    },
+
+    trackClickEvent: function (schema) {
+      window.dataLayer = window.dataLayer || []
+      window.dataLayer.push(schema)
+    },
+
+    hrefPointsToDomain: function (href, domain) {
+      var httpDomain = 'http://' + domain
+      var httpsDomain = 'https://' + domain
+      var schemaRelativeDomain = '//' + domain
+      return this.stringStartsWith(href, domain) ||
+      this.stringStartsWith(href, httpDomain) ||
+      this.stringStartsWith(href, httpsDomain) ||
+      this.stringStartsWith(href, schemaRelativeDomain)
+    },
+
+    stringStartsWith: function (string, stringToFind) {
+      return string.substring(0, stringToFind.length) === stringToFind
+    },
+
+    hrefIsRelative: function (href) {
+      // Checks that a link is relative, but is not a protocol relative url
+      return this.stringStartsWith(href, '/') && !this.stringStartsWith(href, '//')
```

Comment:
If I understand what this is doing correctly, I think a slightly more efficient way of doing it might be:

```
return href[0] === '/' && href[1] !== '/'
```